nexus/src/app/silo.rs

```diff
@@ -376,6 +391,167 @@ impl super::Nexus {
         Ok(Some(db_silo_user))
     }
 
+    // Silo user passwords
+
+    /// Set or invalidate a Silo user's password
+    ///
+    /// If `password` is `UserPassword::Password`, the password is set to the
+    /// requested value.  Otherwise, any existing password is invalidated so
+    /// that it cannot be used for authentication any more.
+    pub async fn local_idp_user_set_password(
```

Comment:
This function will return an error before creating a password hash if either the user or silo does not exist. Could this be used for timing attacks, e.g., to probe for existing users? Perhaps it should unconditionally call `silo_user_password_verify` like `login_local` does.

Reply:
Yeah, another good catch.  I think this is okay because `silo_user_lookup_by_id` does an authz check that you're allowed to modify the user.  Thus, you'll get a fast error unless you have permission to see the named Silo _and_ modify the specified User (i.e., you'll get a fast error if the Silo doesn't exist, or you don't have permission to see it, or the User doesn't exist, or you don't have permission to modify it).

Reply:
Ok, that seems fine.